ConwaysGameOfLife.Tests/GeneratorShould.cs

```diff
@@ -22,5 +24,88 @@ public void FormatUniverse_GivenUniverseOfAllDeadCells()
 
             Assert.Equal(expected, _output.GetWriteLine());
         }
+
+        [Theory]
+        [InlineData("3,3", 3, 3)]
+        [InlineData("10,10", 10, 10)]
+        public void SetUniverseDimensions_GivenValidInput(string input, int gridWidth, int gridLength)
+        {
+            _input.WithReadLine(input);
+            var result = _generator.SetUniverseDimensions(input);
+
+            Assert.True(UniverseHelper.UniversesAreEqual(new Universe(gridWidth, gridLength), result));
+            Assert.Equal(gridWidth, result.GridWidth);
+            Assert.Equal(gridLength, result.GridLength);
+        }
+
+        [Theory]
+        [InlineData("3,3", 3, 3)]

```

Comment:
this is passing because of the test constructor passing in `new Universe(3, 3)`. be careful with using "similar" test cases frequently i.e. mostly using 3x3 universes. it can lead to tests passing when they shouldn't. luckily, this test is a `[Theory]` so you can tell that the overall function isn't working as intended even though the 3x3 test passed